### PR TITLE
Retain globals in function browser when a new function is added

### LINKED ISSUE #29084: Globals being removed
In multi domain fitting (GUI in mantidplot, muon analysis, indirect)

Open GUI and load data
Add a function
Make some things global
Add another functuion
Globals are gone

Add some globals
Remove one of the functions
Globals are gone again

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR makes it so the global parameters are carried over when a new function is added to the function browser. Previously, when switching from a single to composite function the browser was cleared (and hence the globals). 

**To test:**
1. Open the Muon Analysis 2 GUI
2. Load in any data (e.g MUSR62260)
3. Go to the fitting tab
4. Change the simultaneous  fit (so global checkbox is shown)
5. Add a function
6. Pick a global
7. Add a new function, globals should be retained
8. Remove that new function, globals should be retained

Fixes #29084 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/common/src/FunctionTreeView.cpp [hunk 1]
```diff
@@ -1220,6 +1228,8 @@ void FunctionTreeView::addFunctionEnd(int result) {
       return;
   }
   emit functionAdded(QString::fromStdString(f->asString()));
+  setGlobalParameters(globalParameters);
+  emit globalsChanged(globalParameters);
```
Comment (clayton-tom) [inline qt/widgets/common/src/FunctionTreeView.cpp:1232]: What is the difference between using `emit` and just calling the function?

Comment (StephenSmith25) [reply qt/widgets/common/src/FunctionTreeView.cpp:1232]: globalsChanged is marked as a qt signal. If we emit this signal anything connected to it (e.g the presenter) will be notified. This is necessary as changing this value in the view should notify the presenter, which will then update the model with the new global parameters.

**Review by clayton-tom — DISMISSED** [review]
Works well and code looks good 👍

**Comment by StephenSmith25** [comment]
@clayton-tom Could you have another quick glance at this, your review was dismissed when I added release notes.

**Review by clayton-tom — APPROVED** [review]
All still fine 👍

**Pull request merged**